app/scripts/components/folderComponents/addFolderModal.jsx

```diff
@@ -20,83 +19,69 @@ class AddFolderModal extends React.Component {
             <FlatButton
                 label="Cancel"
                 secondary={true}
-                onTouchTap={this.handleClose.bind(this)}/>,
+                onTouchTap={()=>this.closeModal()}/>,
             <FlatButton
                 label="Submit"
                 secondary={true}
                 keyboardFocused={true}
-                onTouchTap={this.handleFolderButton.bind(this)}/>
+                onTouchTap={()=>this.addFolder()}/>
         ];
-
+        let open = this.props.toggleModal && this.props.toggleModal.id === 'addFolder' ? this.props.toggleModal.open : false;
         return (
             <div>
                 <RaisedButton
                     label="Add Folder"
                     labelStyle={{color: '#235F9C'}}
                     style={styles.addFolder}
-                    onTouchTap={this.openModal.bind(this)}/>
+                    onTouchTap={()=>this.openModal()}/>
                 <Dialog
                     style={styles.dialogStyles}
                     contentStyle={this.props.screenSize.width < 580 ? {width: '100%'} : {}}
                     title="Add New Folder"
                     autoDetectWindowHeight={true}
                     actions={actions}
-                    open={this.state.open}
-                    onRequestClose={this.handleClose.bind(this)}>
+                    open={open}
+                    onRequestClose={()=>this.closeModal()}>
                     <form action="#" id="newFolderForm">
                         <TextField
                             style={styles.textStyles}
                             hintText="Folder Name"
                             errorText={this.state.floatingErrorText}
                             floatingLabelText="Folder Name"
-                            id="folderNameText"
+                            ref={(input) => this.folderNameText = input}
```

Comment:
I read that this is becoming the standard way of doing it. I'll dig up the info and send it to you.